sql/expression/function/sqrt_power.go

```diff
@@ -0,0 +1,162 @@
+package function
+
+import (
+	"fmt"
+	"math"
+
+	"gopkg.in/src-d/go-mysql-server.v0/sql/expression"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+)
+
+// Sqrt is a function that returns the square value of the number provided.
+type Sqrt struct {
+	expression.UnaryExpression
+}
+
+// NewSqrt creates a new Sqrt expression.
+func NewSqrt(e sql.Expression) sql.Expression {
+	return &Sqrt{expression.UnaryExpression{Child: e}}
+}
+
+func (s *Sqrt) String() string {
+	return fmt.Sprintf("sqrt(%s)", s.Child.String())
+}
+
+// Type implements the Expression interface.
+func (s *Sqrt) Type() sql.Type {
+	return sql.Float64
+}
+
+// IsNullable implements the Expression interface.
+func (s *Sqrt) IsNullable() bool {
+	return s.Child.IsNullable()
+}
+
+// TransformUp implements the Expression interface.
+func (s *Sqrt) TransformUp(fn sql.TransformExprFunc) (sql.Expression, error) {
+	child, err := s.Child.TransformUp(fn)
+	if err != nil {
+		return nil, err
+	}
+	return fn(NewSqrt(child))
+}
+
+// Eval implements the Expression interface.
+func (s *Sqrt) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	child, err := s.Child.Eval(ctx, row)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if child == nil {
+		return nil, nil
+	}
+
+	child, err = sql.Float64.Convert(child)
+	if err != nil {
+		return nil, nil
+	}
+
+	return computeSqrt(child.(float64))
+}
+
+func computeSqrt(x float64) (float64, error) {
+	return math.Sqrt(x), nil
+}
+
+type powerFuncName int
```

Comment:
Since `power` is a synonym for `pow` (https://dev.mysql.com/doc/refman/8.0/en/mathematical-functions.html#function_pow) I think is not necessary have different constructors depending on the name.

We could have just `NewPower` without `powerFuncName`s and use it for both functions where they are registered.